tests/tools/microServer/uWServer.py

```diff
@@ -690,18 +689,18 @@ def main():
         lookup_key_ = args.lookupkey
         MyHandler.protocol_version = HTTP_VERSION
 
-        if IPConstants.isIPv6(options.ip_address):
+        if IPConstants.isIPv6(args.ip_address):
             print("Server running on IPv6")
             HTTPServer.address_family = socket.AF_INET6
 
-        if options.ssl == "True" or options.ssl == "true":
-            server = SSLServer((IPConstants.getIP(options.ip_address), options.port), MyHandler, options)
+        if args.ssl == "True" or args.ssl == "true":
+            server = SSLServer((IPConstants.getIP(args.ip_address), args.port), MyHandler, args)
         else:
-            server = ThreadingServer((IPConstants.getIP(options.ip_address), options.port), MyHandler, options)
+            server = ThreadingServer((IPConstants.getIP(args.ip_address), args.port), MyHandler, args)
 
         server.timeout = 5
-        print("Started server on port {0}".format(options.port))
-        server_thread = threading.Thread(target=server.serve_forever())
+        print("Started server on port {0}".format(args.port))
+        server_thread = threading.Thread(target=server.serve_forever)
```

Comment:
server.serve_forever() doesn't terminate until the process gets killed (usually by AuTest), since the two types of server we are using in microServer both spawns new threads to work on incoming connections, the main thread will just get stuck on the serve_forever function. Changing the target to the function itself allows an actual thread to be created that runs serve_forever for the server, while the main thread is freed up to do other things like listening for SigINTs; whereas before, there has been cases of SigINT not getting caught because the main microServer thread is busy doing serve_forever stuff.